jobs/build.Jenkinsfile

```diff
@@ -178,24 +178,28 @@ lock(resource: "build-${params.STREAM}") {
             """)
         }
 
+        // Determine parent version/commit information
         def parent_version = ""
         def parent_commit = ""
-        stage('Fetch') {
-            if (s3_stream_dir) {
-                pipeutils.aws_s3_cp_allow_noent("s3://${s3_stream_dir}/releases.json", "tmp/releases.json")
-                if (utils.pathExists("tmp/releases.json")) {
-                    def releases = readJSON file: "tmp/releases.json"
-                    // check if there's a previous release we should use as parent
-                    for (release in releases["releases"].reverse()) {
-                        def commit_obj = release["commits"].find{ commit -> commit["architecture"] == basearch }
-                        if (commit_obj != null) {
-                            parent_commit = commit_obj["checksum"]
-                            parent_version = release["version"]
-                            break
-                        }
+        if (s3_stream_dir) {
+            pipeutils.aws_s3_cp_allow_noent("s3://${s3_stream_dir}/releases.json", "tmp/releases.json")
+            if (utils.pathExists("tmp/releases.json")) {
+                def releases = readJSON file: "tmp/releases.json"
+                // check if there's a previous release we should use as parent
+                for (release in releases["releases"].reverse()) {
+                    def commit_obj = release["commits"].find{ commit -> commit["architecture"] == basearch }
+                    if (commit_obj != null) {
+                        parent_commit = commit_obj["checksum"]
+                        parent_version = release["version"]
+                        break
                     }
                 }
+            }
+        }
```

Comment:
Aside: I know I wrote this way back, but probably about time we factor that out into a script in cosa or fedora-coreos-releng-automation. Or maybe it could be a `cosa buildfetch --parent` which fetches the parent build info and creates a `parent` symlink in `builds/`.

Reply:
> Aside: I know I wrote this way back, but probably about time we factor that out into a script in cosa or fedora-coreos-releng-automation.

I was actually looking at `fedora-coreos-releng-automation`. I almost think the scripts there that we only use in the pipeline we should just pull into COSA itself rather than `git clone`ing that repo for every pipeline run.

Or maybe it could be a `cosa buildfetch --parent` which fetches the parent build info and creates a `parent` symlink in `builds/`.

Yeah I think I prefer this ^^